superset/assets/src/dashboard/components/Dashboard.jsx

```diff
@@ -203,11 +213,40 @@ class Dashboard extends React.PureComponent {
     }
   }
 
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleKeydown);
+  }
+
   // return charts in array
   getAllCharts() {
     return Object.values(this.props.charts);
   }
 
+  handleKeydown(event) {
+    const controlOrCommand = event.ctrlKey || event.metaKey;
+    if (controlOrCommand) {
+      const isK = event.key === 'k' || event.keyCode === 83;
+      if (isK) {
+        this.setState({ showOmni: !this.state.showOmni });
+
+        // Get first input in the modal div
+        document
```

Comment:
I think this needs to be more robust, what if more modals are added? what happens if the selectors are empty and this throws an error? 

ideally you would get a DOM ref via React`s `ref` api.

Reply:
I agree. The `getElementsByClassName` and such approach tends to be brittle especially when the layout and classnames of the library we use for modal dialogs changes on a future upgrade.